### PR TITLE
Bug 52333 - Debugger applies rounding when casting float values to int

### PR DESCRIPTION
**Pull request by DavidKarlas**
Since System.Convert.To, can have different behavior then casting, I removed use of this and always use DynamicCast which does proper casting. In theory performance might suffer but this should be minimal and not noticable by user at all.

### CODE REVIEW

**Comment by slluis** [comment]
GenerateTypeCastDelegate is not doing any kind of caching of the generated method. That looks wrong.

**Comment by DavidKarlas** [comment]
Added caching, it's based on method name which is good enough for all our use cases(casting between primary values)

File: Mono.Debugging.Soft/SoftDebuggerAdaptor.cs [hunk 1]
```diff
@@ -202,53 +202,60 @@ public override object TryConvert (EvaluationContext ctx, object obj, object tar
...
-				PropertyInfo propInfo = fromType.GetProperty ("Value");
-				methodInfo = propInfo.GetGetMethod ();
-
-				il.Emit (OpCodes.Stloc_0);
-				il.Emit (OpCodes.Ldloca_S);
-				il.Emit (OpCodes.Call, methodInfo);
-
-				fromType = methodInfo.ReturnType;
-			}
+			lock(typeCastDelegatesCache) {
```
Comment (Therzok) [inline Mono.Debugging.Soft/SoftDebuggerAdaptor.cs]: NIt: missing space.

Comment (DavidKarlas) [inline Mono.Debugging.Soft/SoftDebuggerAdaptor.cs]: Formatting settings/Roslyn problem

**Pull request merged**